lib/parsers/package-lock-parser.ts

```diff
@@ -41,66 +51,154 @@ export class PackageLockParser implements LockfileParser {
       version: manifestFile.version,
     };
 
-    const topLevelDeps: Dep[] = getTopLevelDeps(manifestFile, includeDev);
-
     // asked to process empty deps
     if (_.isEmpty(manifestFile.dependencies) && !includeDev) {
       return depTree;
     }
 
+    // prepare a flat map, where keys are path, where dependencies are listed
+    const depMap: DepMap = await this.flattenLockfile(lockfile);
```

Comment:
pretty sure all this `async/await` is not needed. given no async IO is happening in this loop, the async/await would still block the even-loop until done

Reply:
Yep, that's one of the thing I need your help and expertise - it's not needed now, but this package can end up being used in a deployed service, so we need to be sure we won't block event loop there. I know (from your talk) can push event loop to tick by `setImmediate()`, but I'd like to discuss the right approach.